sm/plugin_sm/tests/plugin.rs

```diff
@@ -0,0 +1,190 @@
+#[cfg(test)]
+mod tests {
+    // Tests: calls to each of Plugin API commands: prepare, list, install, remove, finalize, version
+    // Check exit codes
+    // Check some sample output
+    // use Dummy plugin
+    // Add crash test due to no timeout plugin may never return
+    // Multiple version of invalid output, eg: JSON Lines use '\r\n' separator, JSON instead, garbage
+    // Try 10000 lines output
+    //
+
+    use json_sm::{SoftwareError, SoftwareModule};
+    use plugin_sm::plugin::{ExternalPluginCommand, Plugin};
+    use std::{fs, io::Write, path::PathBuf, str::FromStr};
+    #[tokio::test]
+    async fn plugin_get_command_prepare() {
+        let (plugin, _plugin_path) = get_dummy_plugin("test");
+
+        let res = plugin.prepare().await;
+
+        assert_eq!(res, Ok(()));
+    }
+
+    #[tokio::test]
+    async fn plugin_get_command_finalize() {
+        let (plugin, _plugin_path) = get_dummy_plugin("test");
+
+        let res = plugin.finalize().await;
+
+        assert_eq!(res, Ok(()));
+    }
+
+    #[tokio::test]
+    async fn plugin_get_command_list() {
+        let (plugin, _plugin_path) = get_dummy_plugin("test");
+        let path = PathBuf::from_str("/tmp/.tedge_dummy_plugin").unwrap();
+        if !&path.exists() {
+            let () = fs::create_dir(&path).unwrap();
+        }
+
+        let mut file = tempfile::Builder::new()
+            .suffix(".0")
+            .tempfile_in(path)
+            .unwrap();
+
+        let content = r#"{"name":"abc","version":"1.0"}"#;
+        let _a = file.write_all(content.as_bytes()).unwrap();
+
+        let res = plugin.list().await;
+        assert!(res.is_ok());
+    }
+
+    #[tokio::test]
+    async fn plugin_get_command_install() {
+        let (plugin, _plugin_path) = get_dummy_plugin("test");
+        let path = get_dummy_plugin_tmp_path();
+
+        let mut file = tempfile::Builder::new()
+            .suffix(".0")
+            .tempfile_in(path)
+            .unwrap();
+
+        let content = r#"{"name":"abc","version":"1.0"}"#;
+        let _a = file.write_all(content.as_bytes()).unwrap();
+
+        let module = SoftwareModule {
+            module_type: Some("test".into()),
+            name: "test".into(),
+            version: None,
+            url: None,
+        };
+        let res = plugin.install(&module).await;
+        assert!(res.is_ok());
```

Comment:
Very minor. It's technically the same as line 30. I like assert_eq as it's easier to understand res is Ok(()).
```
assert_eq!(res, Ok(()));
```

Reply:
If there is no errors the API return only `Ok(())`.